Resources/Storage/StorageFactory.py

```diff
@@ -203,9 +210,34 @@ def _getConfigStorageName( self, storageName ):
       aliasName = gConfig.getValue( configPath )
       result = self._getConfigStorageName( aliasName )
       if not result['OK']:
-        errStr = "StorageFactory._getConfigStorageName: Supplied storage doesn't exist."
-        gLogger.error( errStr, configPath )
-        return S_ERROR( errStr )
+        return result
+      resolvedName = result['Value']
+    else:
+      resolvedName = storageName
+    return S_OK( resolvedName )
+
+  def _getBaseStorageName( self, storageName ):
+    """
+      This gets the name of the base SE in case the SE is defined from a base SE.
+      'storageName' is the storage section to check in the CS
+    """
+    configPath = '%s/%s' % ( self.rootConfigPath, storageName )
+    res = gConfig.getOptions( configPath )
+    if not res['OK']:
+      errStr = "StorageFactory._getConfigStorageName: Failed to get storage options"
+      gLogger.error( errStr, res['Message'] )
+      return S_ERROR( errStr )
+    if not res['Value']:
+      errStr = "StorageFactory._getConfigStorageName: Supplied storage doesn't exist."
+      gLogger.error( errStr, configPath )
+      return S_ERROR( errStr )
+    if 'BaseSE' in res['Value']:
+      configPath = '%s/%s/BaseSE' % ( self.rootConfigPath, storageName )
+      baseName = gConfig.getValue( configPath )
+      # Just in case the base is an alias ;-)
+      result = self._getConfigStorageName( baseName )
+      if not result['OK']:
+        return result
```

Comment:
This new function is just a duplication of code, can be made a single function with an extra argument, e.g. referenceType, that can take "Alias" and "BaseSE" string values. If BaseSE concept is introduced, then eventually the SE definition can take all values from the Base SE and be able to override some of them in its own definition